jina/types/arrays/neural_ops.py

```diff
@@ -53,16 +55,35 @@ def match(
         :param use_scipy: use Scipy as the computation backend
         :param metric_name: if provided, then match result will be marked with this string.
         """
+        is_sparse = False
+        if isinstance(self.get_attributes('embedding')[0], np.ndarray):
```

Comment:
u do not want to get all the attributes to check the  first embedding. Get the first document and its embedding, It would be much more efficient

Reply:
yes, it was bad, updated